flocker/control/_persistence.py

```diff
@@ -306,6 +310,77 @@ def _cached_dfs_serialize(input_object):
 
     return result
 
+# A couple tokens that are used below in the generation hash.
+_NULLSET_TOKEN = mmh3_hash_bytes(b'NULLSET')
+_MAPPING_TOKEN = mmh3_hash_bytes(b'MAPPING')
+_STR_TOKEN = mmh3_hash_bytes(b'STRING')
+
+_generation_hash_cache = WeakKeyDictionary()
+
+
+def generation_hash(input_object):
+    """
+    This computes the mmh3 hash for an input object, providing a consistent
+    hash of deeply persistent objects across python nodes and implementations.
+
+    :returns: An mmh3 hash of input_object.
+    """
+    # Ensure this is a quick function for basic types:
+    # Note that ``type(x) in frozenset([str, int])`` is faster than
+    # ``isinstance(x, (str, int))``.
+    input_type = type(input_object)
+    if (
+            input_object is None or
+            input_type in _BASIC_JSON_TYPES
+    ):
+        if input_type == unicode or input_type == bytes:
+            # Add a token to identify this as a string.
+            object_to_process = b''.join([_STR_TOKEN, bytes(input_object)])
```

Comment:
:question: I don't know what `bytes(u"some unicode string")` does. Which encoding does it use? It might be clearer to say `input_object.encode('utf8')`